### PR TITLE
Classloader leak detection utility for tests

### PR DESCRIPTION
**Pull request by Sanne**
Introducing some little helpers to conveniently verify that certain operations won't leak objects, with some additional tools focusing on not leaking the classloader.

Related to:

https://hibernate.atlassian.net/browse/HHH-16911

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/bootstrap/registry/classloading/ClassloaderLeaksUtilityTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+package org.hibernate.orm.test.bootstrap.registry.classloading;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Verifies basic operations of {@link ClassLoaderLeakDetector}.
+ */
+public class ClassloaderLeaksUtilityTest {
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/bootstrap/registry/classloading/ClassloaderLeaksUtilityTest.java:15]:
```suggestion
public class ClassLoaderLeaksUtilityTest {
```

**Comment by Sanne** [comment]
Rebased again. I'll merge soon if there's no further comments.

**Comment by Sanne** [comment]
Rebased to see CI's output in combination with #7083

**Comment by Sanne** [comment]
Rebased again,  to also incorporate #[6537](https://github.com/hibernate/hibernate-orm/pull/6537)

**Comment by Sanne** [comment]
Ok, I think I might have fixed it - pushed again

**Pull request merged**